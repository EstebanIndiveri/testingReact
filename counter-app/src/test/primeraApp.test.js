import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
// const { render } = require('@testing-library/react');
const { default: PrimeraApp } = require('../PrimeraApp');

describe('Pruebas en primeraApp />', () => {
    // test('hola', () => {
    //     const saludo='hola';
    //     const {getByText}=render(<PrimeraApp saludo={saludo}/>)
        
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })
    test('<PrimeraApp/>', () => {
        const saludo='hola';
        const wrapper=shallow(<PrimeraApp saludo={saludo}/>)

        expect(wrapper).toMatchSnapshot();
    })
    test('debe de mostrar subtitle',()=>{
        const saludo='hola';
        const subtitle='subtitle';
        const wrapper=shallow(
            <PrimeraApp 
            saludo={saludo}
            subtitle={subtitle}/>
        );
        const textParrafo=wrapper.find('p').text();

            expect(textParrafo).toBe(subtitle);
    })
})
