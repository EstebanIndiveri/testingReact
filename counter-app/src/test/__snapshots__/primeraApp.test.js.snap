// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en primeraApp /> <PrimeraApp/> 1`] = `
<Fragment>
  <h1>
    hola
  </h1>
  <p
    style={
      Object {
        "color": "red",
        "textDecoration": "underline",
      }
    }
  >
    subtitle
  </p>
</Fragment>
`;
